src/mscorlib/src/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -732,7 +733,10 @@ private unsafe SortKey CreateSortKey(String source, CompareOptions options)
 
                 fixed (byte* pSortKey = keyData)
                 {
-                    Interop.Globalization.GetSortKey(_sortHandle, source, source.Length, pSortKey, sortKeyLength, options);
+                    if (Interop.Globalization.GetSortKey(_sortHandle, source, source.Length, pSortKey, sortKeyLength, options) != sortKeyLength)
+                    {
+                        throw new ArgumentException(SR.Arg_MustBeString, nameof(source));
```

Comment:
How can `source` not be a string? It's got invalid surrogates? Maybe message should say it has invalid content.

Reply:
And possibly attempt to include the value

Reply:
The error handling for (theoretical?) errors coming out of GetSortKey was fairly inconsistent. It was combination of: `Arg_MustBeString` argument exception, FailFast, and ignoring the errors.

I have standardized on `Arg_MustBeString` because of it is what the original native implementation used: https://github.com/dotnet/coreclr/pull/17029/files#diff-5eb37e957a9809fb3b4d9574f91b8d12L126

However, I agree that this is really confusing message if/ever we hit it. I am changing it to `Arg_ExternalException` that we are using in other places for unexpected errors coming out of globalization APIs.

Reply:
> And possibly attempt to include the value

We are not including the string value for any other errors coming out of globalization. And I am not sure whether including them would be a good idea. If the globalization APIs fail, it will likely be because of they got passed in a weird string. Such strings are likely going to cause bad things to happen when they are printed, and so you end up with worse diagnostic experience.

Reply:
Makes sense.